### PR TITLE
No fallback for Kuadrant

### PR DESCRIPTION
**Pull request by pehala**
- If Kuadrant is enabled, it will fail if it cannot find it.

### CODE REVIEW

**Review by averevki — COMMENTED** [review]
How about some explicit custom exception(s) in case of failure? This will make it clear why tests are failing

**Comment by pehala** [comment]
> How about some explicit custom exception(s) in case of failure? This will make it clear why tests are failing

It now uses pytest.fail with a better message that should make it clear

File: testsuite/tests/conftest.py [hunk 1]
```diff
@@ -204,25 +204,21 @@ def module_label(label):
...
+    kuadrants = kuadrant_openshift.do_action("get", "kuadrant", "-o", "json", parse_output=True)
+    assert len(kuadrants.model["items"]) > 0
+
+    # Try if the configured Gateway is deployed
+    gateway_openshift = openshift.change_project(settings["kuadrant"]["gateway"]["project"] % None)
+    name = testconfig["kuadrant"]["gateway"]["name"]
+    gateway_openshift.do_action("get", f"Gateway/{name}")
+
+    # TODO: Return actual Kuadrant object
+    return True
```
[the remark this answers is not in the document]
Comment (pehala) [reply testsuite/tests/conftest.py:236]: Is it now better?

File: testsuite/tests/conftest.py [hunk 2]
```diff
@@ -248,7 +249,8 @@ def backend(request, openshift, blame, label):
 def envoy(request, kuadrant, authorino, openshift, blame, backend, module_label, testconfig) -> Proxy:
     """Deploys Envoy that wire up the Backend behind the reverse-proxy and Authorino instance"""
     if kuadrant:
-        envoy: Proxy = Gateway(openshift, "istio-ingressgateway", "istio-system", module_label, backend)
+        config = testconfig["kuadrant"]["gateway"]
+        envoy: Proxy = Gateway(openshift, config["name"], config["project"], module_label, backend)
```
Comment (averevki) [inline testsuite/tests/conftest.py:253]: same here

**Review by averevki — APPROVED** [review]

**Review by mganisin — APPROVED** [review]

**Pull request merged**